chia/timelord/timelord.py

```diff
@@ -337,6 +341,8 @@ async def _reset_chains(self, *, first_run: bool = False, only_eos: bool = False
                     self.iteration_to_proof_type[new_block_iters] = IterationType.INFUSION_POINT
         # Remove all unfinished blocks that have already passed.
         self.unfinished_blocks = new_unfinished_blocks
+        # Remove old overflow blocks that we didn't infuse.
```

Comment:
we clean all the blocks we cant infuse in the loop above but we dont clean the blocks that we moved from self.overflow_blocks to the new_unfinished

i think just adding 
if block in self.overflow_blocks:
                        self.overflow_blocks.remove(block)
                        
to line 337 would be a better solution, also it looks like we currently duplicate blocks in both self.unfinished and self.overflow_blocks in the loop above